File: SE1Challenge/SE1Challenge/Services/OrderLogger.swift
//
//  OrderLogger.swift
//  SE1Challenge
//
//  Created by Michal Miko on 19.08.2021.
//

import Foundation

protocol Analytics {
    static func log(order: Order, price: OrderPrice)
}

final class Logger: Analytics {
    static func log(order: Order, price: OrderPrice) {
        print("Order: \(order.id). Price: \(price.price)")
    }
}

File: app/src/main/java/com/sima/se1challenge/order/logging/OrderLogger.kt
package com.sima.se1challenge.order.logging

import com.sima.se1challenge.model.Order
import com.sima.se1challenge.model.OrderPrice

interface OrderLogger {
    fun log(order: Order, orderPrice: OrderPrice)

    companion object {
        fun getDefault(): OrderLogger = OrderLoggerImpl()
    }
}
